### ISSUE TITLE
Switching between local computer and RDP switches NumLock on

### ISSUE THREAD

**Issue by SuperDre** [neon-dev/RDPHotkeyHelper#5]
I always have my NumLock off, but when running RDPHotkeyHelper in the background and switching from my local computer to my RDP computer the NumLock turns on. I can turn it off again, but when I switch back to local and then back to RDP it automatically switches on again.

Steps to reproduce (Windows 10 Pro):
- Local (computer): switch numlock off
- Switch from local to RDP window (I have it fullscreen): the Numlock automatically switches on
- RDP: switch numlock off
- Switch from RDP to local: numlock stays off
- Switch back from local to RDP: numlcok switches on again

When I stop RDPHotkeyHelper and do the steps again, the numlock stays off the whole time.

Edit:
Ahhh I see you do it deliberately:
  _SetNumLockState, On ; the remote desktop client disables num lock upon first connection for some reason_
So I guess it should be optional.
Have an extra menu item 'NumLock: `State`', and have the user select the option to turn it off (default take the current state of the NumLock when RDPHotkeyHelper was started).

**Comment by SuperDre** [comment]
Having played with the script for some time now, I can't reproduce the reason why you added the SetNumLockState. On first connection with my Windows 10 Pro and the RDP that came with it, the NumLock did not get turned off. (RDP starts fullscreen and has Keyboard set to 'Only when using fullscreen').
Maybe the following could be an improvement (without any menu's and the user needing to switch):
(Get the NumLock State on startup, and when it the focuses goes back to local computer)

```
#KeyHistory 0
NumLock := GetKeyState("NumLock", "T")  ;0=off 1=on**
UpdateTrayIcon()
Menu, Tray, Tip, RDP Hotkey Helper
Menu, Tray, NoStandard
Menu, Tray, Add, Exit

global RdpWindowTitle := % (SubStr(A_Language, -1) = "07" ? " - Remotedesktop" : " - Remote Desktop") ; matches both classic RDP client and the app from the Microsoft store
SetTitleMatchMode, 2
DetectHiddenWindows, On
While true {
  WinWaitActive, %RdpWindowTitle%
  Suspend, On ; rebind all hotkeys, otherwise they won't work in RDP fullscreen mode
  Suspend, Off
  Sleep, 250
  SetNumLockState, NumLock ; the remote desktop client disables num lock upon first connection for some reason
  WinWaitNotActive, %RdpWindowTitle%
  NumLock := GetKeyState("NumLock", "T")
}

```

Only problem with above version is that after you switched back to local and then change the NumLock state, it will reset the NumLock back to the state when you last switched from RDP to Local (but I think that is still much less of a problem as the current version with always setting it to On is).

**Comment by neon-dev** [comment]
Thanks for your investigation and suggested solution.
I'll first try to reproduce the behavior for which I introduced this override. If it's not needed anymore we can safely remove it and don't have to worry about state management.

**Comment by neon-dev** [comment]
Fixed (see fc9fc0f)
